src/main/java/com/minecolonies/api/colony/requestsystem/requestable/Tool.java

```diff
@@ -129,45 +161,39 @@ public static void serialize(final IFactoryController controller, final Friendly
      */
     public static Tool deserialize(final IFactoryController controller, final FriendlyByteBuf buffer)
     {
-        final IToolType type = ToolType.getToolType(buffer.readUtf(32767));
+        final ToolTypeEntry type = IToolTypeRegistry.getInstance().getValue(buffer.readResourceLocation());
         final int minLevel = buffer.readInt();
         final int maxLevel = buffer.readInt();
         final ItemStack result = buffer.readBoolean() ? buffer.readItem() : ItemStack.EMPTY;
 
         return new Tool(type, minLevel, maxLevel, result);
     }
 
-    /**
-     * Returns the tool class that is requested.
-     *
-     * @return The tool class that is requested.
-     */
-    @NotNull
-    public IToolType getToolClass()
+    @Override
+    public boolean matches(@NotNull final ItemStack stack)
     {
-        return toolClass;
+        if (ItemStackUtils.isEmpty(stack))
+        {
+            return false;
+        }
+
+        if (!getToolClass().checkIsTool(stack)) {
```

Comment:
This check is unnecessary because `ItemStackUtils#hasToolLevel` checks this as well.

Reply:
Done.